Documentation/ExtensionArchitecture/Tutorials/Tea/Model.rst

```diff
@@ -0,0 +1,289 @@
+..  include:: /Includes.rst.txt
+
+..  index::
+    Tutorial Tea; Model
+..  _extbase_tutorial_tea_model:
+
+===================
+Model: a bag of tea
+===================
+
+We keep the model basic: Each tea can have a title, a description, and an optional image.
+
+..  uml::
+
+    class Tea {
+        title
+        ==
+        description
+        ==
+        image
+    }
+
+The title and description are strings, the image is stored as a relation
+to the model class :php:`\TYPO3\CMS\Extbase\Domain\Model\FileReference`, provided
+by Extbase.
+
+
+..  _extbase_tutorial_tea_model_database:
+
+The database model
+==================
+
+Let us translate this into SQL and store the schema in a file called
+:file:`ext_tables.sql`:
+
+..  include:: /CodeSnippets/Tutorials/Tea/ExtTablesSql.rst.txt
+
+The image is stored as an integer. However the field :sql:`image` in the
+database does not contains a reference to the image in form of an identifier.
+
+The field :sql:`image` keeps track of the number of attached images. A separate table,
+a so-called MM table, stores the actual relationship. Read about the definition
+of this field here: :ref:`extbase_tutorial_tea_model_columns_image`.
+
+..  hint::
+    If you have had a look at a TYPO3 database before, you might wonder:
+    What about all those special fields like :sql:`uid`, :sql:`pid`,
+    :sql:`deleted`, etc? The answer is that TYPO3 will generate them automatically
+    for you.
+
+..  _extbase_tutorial_tea_model_tca:
+
+TCA - Table Configuration Array
+===============================
+
+The TCA tells TYPO3 about the database model. It defines all fields
+containing data and all semantic fields that have a special meaning within
+TYPO3 (like the :sql:`deleted` field used, for soft deleting).
+
+The TCA also defines how the corresponding input fields in the backend should look.
+
+The TCA is a nested PHP array. In this example, we need the the following
+keys on the first level:
+
```

Comment:
I suggest bringing ctrl/types/columns in the order as they appear on this page and link to the sections.